includes/admin/class-wc-admin-importers.php

```diff
@@ -177,6 +187,48 @@ public function post_importer_compatibility() {
 			}
 		}
 	}
+
+	/**
+	 * Ajax callback for importing one batch of products from a CSV.
+	 */
+	public function do_ajax_product_import() {
+		check_ajax_referer( 'wc-product-import', 'security' );
+
+		if ( ! current_user_can( 'edit_products' ) || ! isset( $_POST['file'] ) ) {
+			wp_die( -1 );
+		}
+
+		include_once( WC_ABSPATH . 'includes/import/class-wc-product-csv-importer.php' );
+
+		$file = $_POST['file'];
+		$params = array(
+			'start_pos' => isset( $_POST['position'] ) ? absint( $_POST['position'] ) : 0,
+			'lines' => isset( $_POST['lines'] ) ? absint( $_POST['lines'] ) : 10,
+			'mapping' => isset( $_POST['mapping'] ) ? (array) $_POST['mapping'] : array(),
+			'parse' => true,
+		);
+
+		$importer = new WC_Product_CSV_Importer( $file, $params );
```

Comment:
Should be better apply the follow filter, so people can change the importer class:

```
$importer_class = apply_filters( 'woocommerce_product_csv_importer_class', 'WC_Product_CSV_Importer' );
$importer = new $importer_class( $file, $args );
```